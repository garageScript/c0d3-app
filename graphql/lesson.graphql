query Lesson {
  lessons {
    id
    title
  }
}