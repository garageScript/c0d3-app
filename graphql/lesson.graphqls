type Lesson {
  id: String
  title: String
}

type Query {
  lessons: Lesson!
}