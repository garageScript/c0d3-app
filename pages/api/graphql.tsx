import { ApolloServer, gql } from 'apollo-server-micro'

const typeDefs = gql`
  type Query {
    username: String
    hello: String
  }
`

const resolvers = {
  Query: {
    username(_parent: any, _args: any, ctx: any) {
      return ctx.user
    },
    hello() {
      return 'hello'
    }
  }
}

const context = ({ req }: { req: any }) => req.cookies

const apolloServer = new ApolloServer({ typeDefs, resolvers, context })

export const config = {
  api: {
    bodyParser: false
  }
}

export default apolloServer.createHandler({ path: '/api/graphql' })
