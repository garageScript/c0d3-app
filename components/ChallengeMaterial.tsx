import React, { useState } from 'react'
import {
  Challenge,
  ChallengeSubmissionData,
  UserSubmission,
  UserSubmissionsObject
} from '../@types/challenge'
import { LessonStatus } from '../@types/lesson'
import NavLink from './NavLink'
import Markdown from 'markdown-to-jsx'
import gitDiffParser, { File } from 'gitdiff-parser'
import ReactDiffViewer from 'react-diff-viewer'
import Prism from 'prismjs'
import dayjs from 'dayjs'
import relativeTime from 'dayjs/plugin/relativeTime'
import { GiveStarCard } from '../components/GiveStarCard'
import _ from 'lodash'

dayjs.extend(relativeTime)

type CurrentChallengeID = string | null

type ReviewStatusProps = {
  status: string
  reviewerUserName: string | null
}

export const ReviewStatus: React.FC<ReviewStatusProps> = ({
  status,
  reviewerUserName
}) => {
  let reviewStatusComment
  let statusClassName
  const profileLink = (
    <NavLink
      className="text-reset"
      path="/profile/[username]"
      as={`/profile/${reviewerUserName}`}
    >
      {reviewerUserName}
    </NavLink>
  )
  switch (status) {
    case 'passed':
      reviewStatusComment = (
        <>Your solution was reviewed and accepted by {profileLink}</>
      )
      statusClassName = 'border border-success text-success'
      break
    case 'needMoreWork':
      reviewStatusComment = (
        <>Your solution was reviewed and rejected by {profileLink}</>
      )
      statusClassName = 'border border-danger text-danger'
      break
    case 'open':
      reviewStatusComment = (
        <>Your submission is currently waiting to be reviewed</>
      )
      statusClassName = 'border border-warning text-warning'
      break
    default:
      return null
  }
  return (
    <div className={`text-center p-2 my-2 ${statusClassName}`}>
      {reviewStatusComment}
    </div>
  )
}

type StatusIconProps = {
  status: string
}
const StatusIcon: React.FC<StatusIconProps> = ({ status }) => {
  if (status === 'unsubmitted') {
    return null
  }
  let statusIconUrl
  switch (status) {
    case 'passed':
      statusIconUrl = '/assets/curriculum/icons/checkmark.svg'
      break
    case 'needMoreWork':
      statusIconUrl = '/assets/curriculum/icons/rejected.svg'
      break
    case 'open':
      statusIconUrl = '/assets/curriculum/icons/pending.svg'
  }
  return <img width="25px" height="25px" src={statusIconUrl} />
}

type ChallengeTitleCardProps = {
  key: string
  id: string
  title: string
  challengeNum: number
  submissionStatus: string
  active?: boolean
  setCurrentChallenge: React.Dispatch<CurrentChallengeID>
}

export const ChallengeTitleCard: React.FC<ChallengeTitleCardProps> = ({
  submissionStatus,
  active,
  title,
  id,
  challengeNum,
  setCurrentChallenge
}) => {
  const cardStyles: string[] = ['challenge-title-card']
  if (active) {
    cardStyles.push('challenge-title-card--active')
  }
  if (submissionStatus === 'passed') {
    cardStyles.push('challenge-title-card--done')
  } else {
    cardStyles.push('shadow-sm', 'border-0')
  }

  return (
    <div
      data-testid="challenge-title"
      className={`card mb-2 ${cardStyles.join(' ')}`}
      onClick={() => setCurrentChallenge(id)}
    >
      <div className="card-body d-flex justify-content-between">
        <div>{`${challengeNum}. ${title}`}</div>
        <StatusIcon status={submissionStatus} />
      </div>
    </div>
  )
}

type ChallengeQuestionCardProps = {
  currentChallenge: ChallengeSubmissionData
}
export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = ({
  currentChallenge
}) => {
  const diff = _.get(currentChallenge, 'submission.diff', '')
  const comment = _.get(currentChallenge, 'submission.comment', '')
  const updatedAt = _.get(currentChallenge, 'submission.updatedAt', Date.now())
  const reviewerUserName = _.get(
    currentChallenge,
    'submission.reviewer.username',
    null
  )
  let files = null

  if (diff) files = gitDiffParser.parse(diff)

  const renderFile = ({ hunks, newPath }: File) => {
    const oldValue: String[] = []
    const newValue: String[] = []

    hunks.forEach(hunk => {
      hunk.changes.forEach(change => {
        if (change.isDelete) oldValue.push(change.content)
        else if (change.isInsert) newValue.push(change.content)
        else {
          oldValue.push(change.content)
          newValue.push(change.content)
        }
      })
    })

    const syntaxHighlight = (str: string): any => {
      if (!str) return

      const language = Prism.highlight(
        str,
        Prism.languages.javascript,
        'javascript'
      )
      return <span dangerouslySetInnerHTML={{ __html: language }} />
    }

    return (
      <ReactDiffViewer
        key={_.uniqueId()}
        oldValue={oldValue.join('\n')}
        newValue={newValue.join('\n')}
        renderContent={syntaxHighlight}
        splitView={false}
        leftTitle={`${newPath}`}
      />
    )
  }

  return (
    <>
      <div className="card shadow-sm border-0">
        <div className="card-body challenge-question">
          <h1 data-testid="challenge-question-title" className="card-title">
            {currentChallenge.title}
          </h1>
          <div
            data-testid="challenge-question-description"
            className="bg-light p-3 mt-3"
          >
            <Markdown>{currentChallenge.description}</Markdown>
          </div>
        </div>
      </div>

      {diff && (
        <div className="card shadow-sm border-0 mt-3">
          <div className="card-header bg-white">
            Submitted {dayjs(parseInt(updatedAt)).fromNow()}
          </div>
          <div className="card-body">
            <div className="rounded-lg overflow-hidden">
              {files && files.map(renderFile)}
            </div>
          </div>
          <div className="card-footer bg-white">
            {comment && <Markdown>{comment}</Markdown>}
            <ReviewStatus
              status={currentChallenge.status}
              reviewerUserName={reviewerUserName}
            />
          </div>
        </div>
      )}
    </>
  )
}

type ChallengesCompletedCardProps = {
  lessonId: number
  starGiven: string
  imageSrc: string
  reviewUrl: string
  chatUrl: string
}

export const ChallengesCompletedCard: React.FC<ChallengesCompletedCardProps> = ({
  imageSrc,
  chatUrl,
  reviewUrl,
  starGiven,
  lessonId
}) => {
  const [show, setShow] = useState<boolean>(false)
  const [star, setStar] = useState<string>(starGiven)
  return (
    <>
      <GiveStarCard
        close={() => setShow(false)}
        show={show}
        starGiven={star}
        setStarGiven={setStar}
        lessonId={lessonId}
      />
      <div className="card text-center shadow-sm">
        <div className="card-body">
          <img src={`/assets/curriculum/icons/${imageSrc}`} className="mt-3" />
          <h4 className="card-title mt-2">Congratulations!</h4>
          <p className="success-message">
            You have successfully completed all challenges
          </p>
          <p className="review-message">
            You can help your peers by
            <NavLink path={chatUrl} className="font-weight-bold mx-1" external>
              answering questions
            </NavLink>
            they have in the lesson and
            <NavLink
              path={reviewUrl}
              className="font-weight-bold mx-1"
              external
            >
              reviewing challenge submissions
            </NavLink>
          </p>
        </div>
        <div className="card-footer d-flex bg-primary">
          <p className="text-white mr-3 my-2">
            You can show your appreciation to the user that helped you the most
            by giving them a star
          </p>
          <button
            className="btn btn-light text-primary font-weight-bold ml-auto"
            onClick={() => setShow(true)}
          >
            Give Star
          </button>
        </div>
      </div>
    </>
  )
}

type ChallengeMaterialProps = {
  challenges: Challenge[]
  userSubmissions: UserSubmission[]
  lessonStatus: LessonStatus
  chatUrl: string
  lessonId: number
}

const ChallengeMaterial: React.FC<ChallengeMaterialProps> = ({
  challenges,
  userSubmissions,
  lessonStatus,
  chatUrl,
  lessonId
}) => {
  if (!challenges.length) {
    return <h1>No Challenges for this lesson</h1>
  }
  //create an object to evaluate the student's status with a challenge
  const userSubmissionsObject: UserSubmissionsObject = userSubmissions.reduce(
    (acc: UserSubmissionsObject, submission: UserSubmission) => {
      acc[submission.challengeId] = submission
      return acc
    },
    {}
  )
  //chain sort-map causes errors in tests without a copy
  const challengesCopy = [...challenges]
  //create a new Challenges array with user submission data integrated and sorted
  const challengesWithSubmissionData: ChallengeSubmissionData[] = challengesCopy
    .sort((a, b) => a.order - b.order)
    .map((challenge: Challenge) => {
      const submission = userSubmissionsObject[challenge.id] || {}
      return {
        ...challenge,
        status: submission.status || 'unsubmitted',
        submission
      }
    })
  const [
    currentChallengeID,
    setCurrentChallenge
  ] = useState<CurrentChallengeID>(null)

  const finalChallenge = {
    title: 'Challenges Completed!',
    id: 'finalChallenge',
    order: challenges.length + 1,
    description:
      'Congratulations, you have completed all Challenges for this Lesson',
    status: 'passed'
  }
  //find first challenge that is not passed on initial render after clicks will render clicked challenge
  const currentChallenge =
    challengesWithSubmissionData.find((challenge: ChallengeSubmissionData) => {
      if (currentChallengeID) return challenge.id === currentChallengeID
      return challenge.status !== 'passed'
    }) || finalChallenge
  const challengeTitleCards: React.ReactElement[] = challengesWithSubmissionData.map(
    challenge => {
      return (
        <ChallengeTitleCard
          key={challenge.id}
          id={challenge.id}
          challengeNum={challenge.order}
          title={challenge.title}
          setCurrentChallenge={setCurrentChallenge}
          active={challenge.id === currentChallenge.id}
          submissionStatus={challenge.status}
        />
      )
    }
  )
  return (
    <div className="row challenge-display mt-3">
      <div className="col-4">
        {challengeTitleCards}
        {lessonStatus.isPassed && (
          <ChallengeTitleCard
            key={finalChallenge.id}
            id={finalChallenge.id}
            challengeNum={finalChallenge.order}
            title={finalChallenge.title}
            setCurrentChallenge={setCurrentChallenge}
            active={finalChallenge.id === currentChallenge.id}
            submissionStatus={finalChallenge.status}
          />
        )}
      </div>
      <div className="col-8">
        {currentChallenge.id !== 'finalChallenge' && (
          <ChallengeQuestionCard currentChallenge={currentChallenge} />
        )}
        {lessonStatus.isPassed && currentChallenge.id === 'finalChallenge' && (
          <ChallengesCompletedCard
            lessonId={lessonId}
            starGiven={lessonStatus.starGiven || ''}
            imageSrc="icon-challenge-complete.jpg"
            chatUrl={chatUrl}
            reviewUrl={`https://www.c0d3.com/review/${lessonId}`}
          />
        )}
      </div>
    </div>
  )
}

export default ChallengeMaterial
