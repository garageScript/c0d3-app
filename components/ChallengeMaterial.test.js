import React from 'react'
import dayjs from 'dayjs'
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import ChallengeMaterial from './ChallengeMaterial'
import SET_STAR from '../graphql/queries/setStar'
import GET_LESSON_MENTORS from '../graphql/queries/getLessonMentors'
import lessonMentorsData from '../__dummy__/getLessonMentorsData'
import { MockedProvider } from '@apollo/client/testing'
import '@testing-library/jest-dom'
import { SubmissionStatus } from '../graphql'

const mocks = [
  {
    request: {
      query: SET_STAR,
      variables: { mentorId: 4, lessonId: 5, comment: '1' }
    },
    result: {
      data: { setStar: { success: true } }
    }
  },
  {
    request: { query: GET_LESSON_MENTORS, variables: { lessonId: '5' } },
    result: {
      data: { getLessonMentors: lessonMentorsData }
    }
  }
]

const lessonStatusNoPass = {
  isEnrolled: '213423534',
  isTeaching: null,
  starGiven: ''
}

const challenges = [
  {
    id: '105',
    description:
      'Write a function that takes in a number and returns true if that number is greater than 5. Otherwise, return false.',
    title: 'Greater than 5',
    order: 0,
    __typename: 'Challenge'
  },
  {
    id: '107',
    description:
      'Write a function that takes in 2 numbers and returns their sum.',
    title: 'Sum of 2 Numbers',
    order: 1,
    __typename: 'Challenge'
  }
]

const userSubmissions = [
  {
    id: '3500',
    status: SubmissionStatus.Open,
    mrUrl: 'github.com/testmrurl',
    diff:
      'diff --git a/curriculum/js0/2.js b/curriculum/js0/2.js\nindex 647ca32..ac44196 100644\n--- a/curriculum/js0/2.js\n+++ b/curriculum/js0/2.js\n@@ -7,7 +7,7 @@\n  */\n \n const solution = (a, b, c) => {\n-  return 0;\n+  return a + b + c;\n };\n \n module.exports = {\n',
    viewCount: 0,
    comment: null,
    challengeId: '105',
    reviewer: null,
    createdAt: '1586907809223',
    updatedAt: dayjs().subtract(16, 'day').valueOf()
  },
  {
    id: '3501',
    status: SubmissionStatus.NeedMoreWork,
    mrUrl: 'github.com/testmrurl2',
    diff:
      'diff --git a/curriculum/js0/2.js b/curriculum/js0/2.js\nindex 647ca32..ac44196 100644\n--- a/curriculum/js0/2.js\n+++ b/curriculum/js0/2.js\n@@ -7,7 +7,7 @@\n  */\n \n const solution = (a, b, c) => {\n-  return 0;\n+  return a + b + c;\n };\n \n module.exports = {\n',
    viewCount: 0,
    comment: 'test comment',
    challengeId: '107',
    reviewer: {
      id: '1',
      username: 'dan'
    },
    createdAt: '1586907809223',
    updatedAt: dayjs().subtract(16, 'day').valueOf()
  },
  {
    id: '3502',
    status: null,
    mrUrl: 'github.com/testmrurl',
    diff:
      'diff --git a/curriculum/js0/2.js b/curriculum/js0/2.js\nindex 647ca32..ac44196 100644\n--- a/curriculum/js0/2.js\n+++ b/curriculum/js0/2.js\n@@ -7,7 +7,7 @@\n  */\n \n const solution = (a, b, c) => {\n-  return 0;\n+  return a + b + c;\n };\n \n module.exports = {\n',
    viewCount: 0,
    comment: null,
    challengeId: '105',
    reviewer: null,
    createdAt: '1586907809223',
    updatedAt: dayjs().subtract(16, 'day').valueOf()
  }
]

describe('Curriculum challenge page', () => {
  let props
  const setShow = jest.fn()
  beforeEach(() => {
    jest.clearAllMocks()
    props = {
      challenges,
      lessonStatus: lessonStatusNoPass,
      userSubmissions,
      chatUrl: 'https://chat.c0d3.com/c0d3/channels/js0-foundations',
      lessonId: '5',
      setShow
    }
  })

  test('Should render appropriately when no challenges are passed to component', async () => {
    props.challenges = []
    props.userSubmissions = []
    const { container } = render(<ChallengeMaterial {...props} />)
    expect(container).toMatchSnapshot()
  })

  test('Should render first challenge by default when user has no submissions', async () => {
    props.userSubmissions = []
    const { container } = render(<ChallengeMaterial {...props} />)
    expect(container).toMatchSnapshot()
  })

  test('Should render clicked challenge within challenge question', async () => {
    const { getAllByTestId, container } = render(
      <ChallengeMaterial {...props} />
    )
    const challengeTitleCard = getAllByTestId('challenge-title')[1]
    fireEvent.click(challengeTitleCard)
    expect(container).toMatchSnapshot()
  })

  test('Should render first challenge that is not passed when user has submissions', async () => {
    const { container } = render(<ChallengeMaterial {...props} />)
    expect(container).toMatchSnapshot()
  })

  test('Should render challenge material page differently when user has passed all their challenges', async () => {
    const { lessonStatus, userSubmissions } = props
    lessonStatus.isPassed = 'cmon bruh ive passed already'
    userSubmissions.forEach(
      submission => (submission.status = SubmissionStatus.Passed)
    )
    const { container, getByRole, queryByText } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ChallengeMaterial {...props} />
      </MockedProvider>
    )
    expect(container).toMatchSnapshot()

    await waitFor(() =>
      fireEvent.click(getByRole('button', { name: 'Give Star' }))
    )
    await waitFor(() => queryByText('Who helped you the most?'))
    expect(document.body).toMatchSnapshot()

    // click exit button of GiveStarCard
    fireEvent.click(getByRole('img'))
    expect(document.body).toMatchSnapshot()
  })
  test('Should hide mobile modal on click', async () => {
    global.window.innerWidth = 500
    const { container } = render(
      <ChallengeMaterial {...{ ...props, show: true }} />
    )
    expect(screen.getByTestId('modal-challenges')).toBeVisible()
    expect(container).toMatchSnapshot()

    fireEvent.click(screen.getByText('0. Greater than 5'), {
      target: { innerText: '0. Greater than 5' }
    })
    expect(setShow).toBeCalledWith(false)
  })
  test('Should hide mobile modal by clicking on the background', async () => {
    global.window.innerWidth = 500
    const { container } = render(
      <ChallengeMaterial {...{ ...props, show: true }} />
    )
    expect(screen.getByTestId('modal-challenges')).toBeVisible()
    fireEvent.keyDown(container, {
      key: 'Escape',
      code: 'Escape',
      keyCode: 27,
      charCode: 27
    })
    expect(setShow).toBeCalledWith(false)
  })
})
