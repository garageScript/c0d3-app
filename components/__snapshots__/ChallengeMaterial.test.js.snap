// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Curriculum challenge page Should render appropriately when no challenges are passed to component 1`] = `
<div>
  <h1>
    No Challenges for this lesson
  </h1>
</div>
`;

exports[`Curriculum challenge page Should render challenge material page differently when user has passed all their challenges 1`] = `
<div>
  <div
    class="row challenge-display mt-3"
  >
    <div
      class="col-4"
    >
      <div
        class="card mb-2 challenge-title-card challenge-title-card--done"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            0. Greater than 5
          </div>
          <img
            height="25px"
            src="/curriculumAssets/icons/checkmark.svg"
            width="25px"
          />
        </div>
      </div>
      <div
        class="card mb-2 challenge-title-card challenge-title-card--done"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            1. Sum of 2 Numbers
          </div>
          <img
            height="25px"
            src="/curriculumAssets/icons/checkmark.svg"
            width="25px"
          />
        </div>
      </div>
      <div
        class="card mb-2 challenge-title-card challenge-title-card--active challenge-title-card--done"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            3. Challenges Completed!
          </div>
          <img
            height="25px"
            src="/curriculumAssets/icons/checkmark.svg"
            width="25px"
          />
        </div>
      </div>
    </div>
    <div
      class="col-8"
    >
      <div
        class="card text-center shadow-sm"
      >
        <div
          class="card-body"
        >
          <img
            class="mt-3"
            src="/curriculumAssets/icons/icon-challenge-complete.jpg"
          />
          <h4
            class="card-title mt-2"
          >
            Congratulations!
          </h4>
          <p
            class="success-message"
          >
            You have successfully completed all challenges
          </p>
          <p
            class="review-message"
          >
            You can help your peers by
            <a
              class="font-weight-bold mx-1"
              href="https://chat.c0d3.com/c0d3/channels/js0-foundations"
              rel="noopener noreferrer"
              target="_blank"
            >
              answering questions
            </a>
            they have in the lesson and
            <a
              class="font-weight-bold mx-1"
              href="https://www.c0d3.com/review/5"
              rel="noopener noreferrer"
              target="_blank"
            >
              reviewing challenge submissions
            </a>
          </p>
        </div>
        <div
          class="card-footer d-flex bg-primary"
        >
          <p
            class="text-white mr-3 my-2"
          >
            You can show your appreciation to the user that helped you the most by giving them a star
          </p>
          <button
            class="btn btn-light text-primary font-weight-bold ml-auto"
          >
            Give Star
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Curriculum challenge page Should render clicked challenge within challenge question 1`] = `
<div>
  <div
    class="row challenge-display mt-3"
  >
    <div
      class="col-4"
    >
      <div
        class="card mb-2 challenge-title-card shadow-sm border-0"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            0. Greater than 5
          </div>
        </div>
      </div>
      <div
        class="card mb-2 challenge-title-card challenge-title-card--active shadow-sm border-0"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            1. Sum of 2 Numbers
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-8"
    >
      <div
        class="card shadow-sm border-0"
      >
        <div
          class="card-body challenge-question"
        >
          <h1
            class="card-title"
            data-testid="challenge-question-title"
          >
            Sum of 2 Numbers
          </h1>
          <div
            class="bg-light p-3 mt-3"
            data-testid="challenge-question-description"
          >
            <span>
              Write a function that takes in 2 numbers and returns their sum.
            </span>
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Curriculum challenge page Should render first challenge by default when user has no submissions 1`] = `
<div>
  <div
    class="row challenge-display mt-3"
  >
    <div
      class="col-4"
    >
      <div
        class="card mb-2 challenge-title-card challenge-title-card--active shadow-sm border-0"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            0. Greater than 5
          </div>
        </div>
      </div>
      <div
        class="card mb-2 challenge-title-card shadow-sm border-0"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            1. Sum of 2 Numbers
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-8"
    >
      <div
        class="card shadow-sm border-0"
      >
        <div
          class="card-body challenge-question"
        >
          <h1
            class="card-title"
            data-testid="challenge-question-title"
          >
            Greater than 5
          </h1>
          <div
            class="bg-light p-3 mt-3"
            data-testid="challenge-question-description"
          >
            <span>
              Write a function that takes in a number and returns true if that number is greater than 5. Otherwise, return false.
            </span>
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Curriculum challenge page Should render first challenge that is not passed when user has submissions 1`] = `
<div>
  <div
    class="row challenge-display mt-3"
  >
    <div
      class="col-4"
    >
      <div
        class="card mb-2 challenge-title-card challenge-title-card--active shadow-sm border-0"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            0. Greater than 5
          </div>
          <img
            height="25px"
            src="/curriculumAssets/icons/pending.svg"
            width="25px"
          />
        </div>
      </div>
      <div
        class="card mb-2 challenge-title-card shadow-sm border-0"
        data-testid="challenge-title"
      >
        <div
          class="card-body d-flex justify-content-between"
        >
          <div>
            1. Sum of 2 Numbers
          </div>
          <img
            height="25px"
            src="/curriculumAssets/icons/rejected.svg"
            width="25px"
          />
        </div>
      </div>
    </div>
    <div
      class="col-8"
    >
      <div
        class="card shadow-sm border-0"
      >
        <div
          class="card-body challenge-question"
        >
          <h1
            class="card-title"
            data-testid="challenge-question-title"
          >
            Greater than 5
          </h1>
          <div
            class="bg-light p-3 mt-3"
            data-testid="challenge-question-description"
          >
            <span>
              Write a function that takes in a number and returns true if that number is greater than 5. Otherwise, return false.
            </span>
          </div>
        </div>
      </div>
      <div
        class="card shadow-sm border-0 mt-3"
      >
        <div
          class="card-header bg-white"
        >
          Submitted 
          16 days ago
        </div>
        <div
          class="card-body"
        >
          <div
            class="rounded-lg overflow-hidden"
          >
            <table
              class="css-1n5o7vh-diff-container"
            >
              <tbody>
                <tr>
                  <td
                    class="css-17uc4if-title-block"
                    colspan="4"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    >
                      curriculum/js0/2.js
                    </pre>
                  </td>
                </tr>
                <tr
                  class="css-1n7ec1i-line"
                >
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      1
                    </pre>
                  </td>
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      1
                    </pre>
                  </td>
                  <td
                    class="css-17vezug-marker"
                  >
                    <pre />
                  </td>
                  <td
                    class="css-vl0irh-content"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    >
                      <span>
                         
                        <span
                          class="token operator"
                        >
                          *
                        </span>
                        <span
                          class="token operator"
                        >
                          /
                        </span>
                      </span>
                    </pre>
                  </td>
                </tr>
                <tr
                  class="css-1n7ec1i-line"
                >
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      2
                    </pre>
                  </td>
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      2
                    </pre>
                  </td>
                  <td
                    class="css-17vezug-marker css-10regm7-empty-line"
                  >
                    <pre />
                  </td>
                  <td
                    class="css-vl0irh-content css-10regm7-empty-line"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    />
                  </td>
                </tr>
                <tr
                  class="css-1n7ec1i-line"
                >
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      3
                    </pre>
                  </td>
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      3
                    </pre>
                  </td>
                  <td
                    class="css-17vezug-marker"
                  >
                    <pre />
                  </td>
                  <td
                    class="css-vl0irh-content"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    >
                      <span>
                        <span
                          class="token keyword"
                        >
                          const
                        </span>
                         
                        <span
                          class="token function-variable function"
                        >
                          solution
                        </span>
                         
                        <span
                          class="token operator"
                        >
                          =
                        </span>
                         
                        <span
                          class="token punctuation"
                        >
                          (
                        </span>
                        <span
                          class="token parameter"
                        >
                          a
                          <span
                            class="token punctuation"
                          >
                            ,
                          </span>
                           b
                          <span
                            class="token punctuation"
                          >
                            ,
                          </span>
                           c
                        </span>
                        <span
                          class="token punctuation"
                        >
                          )
                        </span>
                         
                        <span
                          class="token operator"
                        >
                          =&gt;
                        </span>
                         
                        <span
                          class="token punctuation"
                        >
                          {
                        </span>
                      </span>
                    </pre>
                  </td>
                </tr>
                <tr
                  class="css-1n7ec1i-line"
                >
                  <td
                    class="css-1ks9aww-gutter css-rq9a2a-diff-removed"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      4
                    </pre>
                  </td>
                  <td
                    class="css-1ks9aww-gutter css-1klnsbn-empty-gutter css-rq9a2a-diff-removed"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    />
                  </td>
                  <td
                    class="css-17vezug-marker css-rq9a2a-diff-removed"
                  >
                    <pre>
                      -
                    </pre>
                  </td>
                  <td
                    class="css-vl0irh-content css-rq9a2a-diff-removed"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    >
                      <span
                        class="css-cncyp1-word-diff"
                      >
                        <span>
                            
                          <span
                            class="token keyword"
                          >
                            return
                          </span>
                           
                        </span>
                      </span>
                      <span
                        class="css-cncyp1-word-diff css-hf3w1f-word-removed"
                      >
                        <span>
                          <span
                            class="token number"
                          >
                            0
                          </span>
                        </span>
                      </span>
                      <span
                        class="css-cncyp1-word-diff"
                      >
                        <span>
                          <span
                            class="token punctuation"
                          >
                            ;
                          </span>
                        </span>
                      </span>
                    </pre>
                  </td>
                </tr>
                <tr
                  class="css-1n7ec1i-line"
                >
                  <td
                    class="css-1ks9aww-gutter css-1klnsbn-empty-gutter css-cnnxkz-diff-added"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    />
                  </td>
                  <td
                    class="css-1ks9aww-gutter css-cnnxkz-diff-added"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      4
                    </pre>
                  </td>
                  <td
                    class="css-17vezug-marker css-cnnxkz-diff-added"
                  >
                    <pre>
                      +
                    </pre>
                  </td>
                  <td
                    class="css-vl0irh-content css-cnnxkz-diff-added"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    >
                      <span
                        class="css-cncyp1-word-diff"
                      >
                        <span>
                            
                          <span
                            class="token keyword"
                          >
                            return
                          </span>
                           
                        </span>
                      </span>
                      <span
                        class="css-cncyp1-word-diff css-1u4zuq6-word-added"
                      >
                        <span>
                          a 
                          <span
                            class="token operator"
                          >
                            +
                          </span>
                           b 
                          <span
                            class="token operator"
                          >
                            +
                          </span>
                           c
                        </span>
                      </span>
                      <span
                        class="css-cncyp1-word-diff"
                      >
                        <span>
                          <span
                            class="token punctuation"
                          >
                            ;
                          </span>
                        </span>
                      </span>
                    </pre>
                  </td>
                </tr>
                <tr
                  class="css-1n7ec1i-line"
                >
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      5
                    </pre>
                  </td>
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      5
                    </pre>
                  </td>
                  <td
                    class="css-17vezug-marker"
                  >
                    <pre />
                  </td>
                  <td
                    class="css-vl0irh-content"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    >
                      <span>
                        <span
                          class="token punctuation"
                        >
                          }
                        </span>
                        <span
                          class="token punctuation"
                        >
                          ;
                        </span>
                      </span>
                    </pre>
                  </td>
                </tr>
                <tr
                  class="css-1n7ec1i-line"
                >
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      6
                    </pre>
                  </td>
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      6
                    </pre>
                  </td>
                  <td
                    class="css-17vezug-marker css-10regm7-empty-line"
                  >
                    <pre />
                  </td>
                  <td
                    class="css-vl0irh-content css-10regm7-empty-line"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    />
                  </td>
                </tr>
                <tr
                  class="css-1n7ec1i-line"
                >
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      7
                    </pre>
                  </td>
                  <td
                    class="css-1ks9aww-gutter"
                  >
                    <pre
                      class="css-h84nzi-line-number"
                    >
                      7
                    </pre>
                  </td>
                  <td
                    class="css-17vezug-marker"
                  >
                    <pre />
                  </td>
                  <td
                    class="css-vl0irh-content"
                  >
                    <pre
                      class="css-o1u8iu-content-text"
                    >
                      <span>
                        module
                        <span
                          class="token punctuation"
                        >
                          .
                        </span>
                        exports 
                        <span
                          class="token operator"
                        >
                          =
                        </span>
                         
                        <span
                          class="token punctuation"
                        >
                          {
                        </span>
                      </span>
                    </pre>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="card-footer bg-white"
        >
          <div
            class="text-center p-2 my-2 border border-warning text-warning"
          >
            Your submission is currently waiting to be reviewed
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
